src/database/schema/UserSchema.ts

```diff
@@ -347,15 +388,15 @@ class UserSchema implements BaseDbSchema {
     return new UserSchema({
       id: uuidv4(),
       channelId,
-      userId: object?.user_id,
+      userId: object?.user?.id || object?.user_id,
       bio: object?.bio,
       countryCode: object?.country_code,
-      createdAt: object?.created_at,
-      lastActiveAt: object?.last_active_at,
-      profilePicture: object?.profile_pic_path,
+      createdAt: object?.user?.created_at,
+      lastActiveAt: object?.last_active_at || new Date().toISOString(),
+      profilePicture: object?.user?.image || DEFAULT_PROFILE_PIC_PATH,
       updatedAt: object?.updated_at,
-      username: object?.username,
-      isBanned: object?.is_banned,
+      username: object?.user?.username,
+      isBanned: object?.user?.banned,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, it's important to check if the properties exist before accessing them. This will prevent potential issues with undefined values.

```diff
- userId: object?.user?.id || object?.user_id,
+ userId: (object?.user && object.user.hasOwnProperty('id')) ? object.user.id : (object.hasOwnProperty('user_id') ? object.user_id : null),

- bio: object?.bio,
+ bio: object.hasOwnProperty('bio') ? object.bio : null,

- countryCode: object?.country_code,
+ countryCode: object.hasOwnProperty('country_code') ? object.country_code : null,

- createdAt: object?.user?.created_at,
+ createdAt: (object?.user && object.user.hasOwnProperty('created_at')) ? object.user.created_at : null,

- lastActiveAt: object?.last_active_at || new Date().toISOString(),
+ lastActiveAt: object.hasOwnProperty('last_active_at') ? object.last_active_at : new Date().toISOString(),

- profilePicture: object?.user?.image || DEFAULT_PROFILE_PIC_PATH,
+ profilePicture: (object?.user && object.user.hasOwnProperty('image')) ? object.user.image : DEFAULT_PROFILE_PIC_PATH,

- updatedAt: object?.updated_at,
+ updatedAt: object.hasOwnProperty('updated_at') ? object.updated_at : null,

- username: object?.user?.username,
+ username: (object?.user && object.user.hasOwnProperty('username')) ? object.user.username : null,

- isBanned: object?.user?.banned,
+ isBanned: (object?.user && object.user.hasOwnProperty('banned')) ? object.user.banned : null,
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->